### ISSUE TITLE
Not working anymore because of Facebook required permissions

### ISSUE THREAD

**Issue by mchiz** [simmatrix/facebook-leads-google-sheets-integration#5]
Hey! I have been using your script for a few months and it worked perfectly. However, a few weeks ago it started failing. I'm getting this error when I'm trying to create a test lead:

CRM access has been revoked from Lead Access Manager

and this is the advice to fix it:

Please make sure your CRM is added in the Lead Access Manager setting for the page (https://www.facebook.com/business/help/1440176552713521)

But I can't see how I can add my app as a CRM. Have you faced this problem?

**Comment by mchiz** [comment]
Update without the solution yet: it seems that now is required to be a verified business. I'm having problems to become verified.

For now, I'm trying to make work the new CRM integration "Google Sheets" (the one that is not from Zapier). It is providing me the leads that the form has already collected, but it is not updating the spreadsheet with the new ones.

I will come back with news.

**Comment by mchiz** [comment]
After more tried, it worked. The bad news is that I'm not sure how... if I have more news, I will come back to post them.

**Comment by Rose2395** [comment]
> Update without the solution yet: it seems that now is required to be a verified business. I'm having problems to become verified.
> 
> For now, I'm trying to make work the new CRM integration "Google Sheets" (the one that is not from Zapier). It is providing me the leads that the form has already collected, but it is not updating the spreadsheet with the new ones.
> 
> I will come back with news.

You do not need to have a verified Business Manager to setup a webhook or to receive leads in real time.

**Comment by Rose2395** [comment]
> After more tried, it worked. The bad news is that I'm not sure how... if I have more news, I will come back to post them.

This might be an issue with the Token that you were using. You can always check your token status here and it is life span.

**Comment by lehieu76** [comment]
Does this script still work? I'm having trouble with the Approval issue

**Comment by Rose2395** [comment]
@lehieu76 , Yes it does work. Can you be mor specific with the issue you are facing or post a screenshot of the error, so we can help you.

**Comment by lehieu76** [comment]
@Rose2395 Thank you for your quick response!
Because Facebook changed the interface, when I created the App I couldn't find the Start a Submission button

I had to go through a complicated process to get Approved

**Can you make a short video to guide me?**

_**Below is an image of what I'm having trouble with.**_
Facebook asked me for the app URL and login account
While pushing data via Google Sheet does not have a URL and Login?

![image](https://github.com/simmatrix/facebook-leads-google-sheets-integration/assets/92743260/c8194abe-ddb3-44eb-a402-d5fe5ebfeb33)

![image](https://github.com/simmatrix/facebook-leads-google-sheets-integration/assets/92743260/6c14abb4-5049-4323-b744-9b4c1da4d0e4)

**Comment by Rose2395** [comment]
@lehieu76 , you do not have to verify the app. Which objective did you choose while creating the app? It should be _other_ and then _Business_ type. I can also make a video for you if that helps you.

**Comment by lehieu76** [comment]
@Rose2395 I chose again and started on the right path
However, I can't find the button **Start a Submission**

Please support me!
![image](https://github.com/simmatrix/facebook-leads-google-sheets-integration/assets/92743260/57e875c4-29f3-44ab-89d5-30511f3e8743)

**Comment by Rose2395** [comment]
@lehieu76 , you do not have to do anything in the App review section.  You only need to do the following in the app interface.

1- Go to App Setting then Basic.  Provide contact email , privacy policy page of your website in the  privacy policy url field, and choose Business and pages as category.

2- Go to webhook section and select page from the dropdown menu. Subscribe for leadgen field, and then click on edit subscription and provide google app script url.

3- On the top, enable the app mode from development to Live.

**Comment by lehieu76** [comment]
@Rose2395 Thank you for helping me get on the right track!

**Instruction 1:** Go to the Add a Permission selection list, click on the subsection **Events Groups Pages**, then tick on the **manage_pages** checkbox
Because Facebook changed the interface, I couldn't find **manage_pages**. Can I tick all?
All available checkboxes are:
pages_manage_cta
pages_manage_instant_articles
pages_show_list
read_page_mailboxes
pages_messaging
pages_messaging_subscriptions
page_events
pages_read_engagement
pages_manage_metadata
pages_read_user_content
pages_manage_ads
pages_manage_posts
pages_manage_engagement
ads_management
ads_read
business_management
instagram_manage_events


**Instruction 2:** Key in me/accounts to the text field beside the Submit button, and make sure you have pasted your long-lived user access token into the Access Token 
I only get blank data 
{
  "data": [
  ]
}

Please guide me. Thank you for your time!

![image](https://github.com/simmatrix/facebook-leads-google-sheets-integration/assets/92743260/d6eeae1e-25ea-411a-83a8-ce4be11646b4)

**Comment by Rose2395** [comment]
@lehieu76 , select your application from the first dropdown and then grant these permissions and generate access token
pages_show_list
ads_management
ads_read
leads_retrieval
pages_read_engagement
pages_manage_data
pages_manage_ads

Then from the second drop down select user token and then again click on generate access token, you will see this window in this step
<img width="325" alt="lead app example" src="https://github.com/simmatrix/facebook-leads-google-sheets-integration/assets/48919446/410a1ac8-c860-45d8-af31-51d752d94fa4">

Click on the edit previous setting and select the page you would like to connect with this app.

Now you should be able to see the page in the User or Page dropdown and your query should return the data.

**Comment by lehieu76** [comment]
I'm really sorry for wasting your time. You have helped me go step by step and get closer to the finish line
I have created the Access token Never Expires
However, I am getting an error in Step 7 (photo below)
Please help me!

![image](https://github.com/simmatrix/facebook-leads-google-sheets-integration/assets/92743260/d6a7f50c-c19f-46d3-9f9e-4fcb4b16d469)
![image](https://github.com/simmatrix/facebook-leads-google-sheets-integration/assets/92743260/96871c48-d076-4a22-ac50-ea2fd3a533f0)

**Comment by Rose2395** [comment]
@lehieu76 , did you manage to get your page in the _User or page_ dropdown menu?

**Comment by lehieu76** [comment]
> @lehieu76 , did you manage to get your page in the _User or page_ dropdown menu?

@Rose2395  I choose Page. 
Can you help me via Teamviewer, Anyviewer or Chrome Remote Desktop?

![image](https://github.com/simmatrix/facebook-leads-google-sheets-integration/assets/92743260/60560a4c-b1cc-496f-9cf8-980e191865e7)

**Comment by Rose2395** [comment]
I can help you on a google meet. Send me an email on this address mehroseyoc@gmail.com
